cmd/exchanger/main.go

```diff
@@ -9,53 +9,52 @@ import (
 
 	"github.com/ethanjmarchand/exchanger/internal/controller"
 	"github.com/ethanjmarchand/exchanger/internal/currency"
-	"github.com/go-chi/chi"
-	"github.com/go-chi/chi/middleware"
 	"github.com/joho/godotenv"
 )
 
-// TODO: Test this funtion.
-// Mixing concerns. This should only load an Env variable. Should be returning a string, or config struct.
-func loadEnvKey() (string, error) {
+// loadEnv simple grabs our env variables from our .env file, and returns a config.
+func loadEnv() (*currency.Config, error) {
+	config := currency.Config{}
 	err := godotenv.Load()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	CCKey := os.Getenv("CCKEY")
-	// Check to see if CCKEY is empty string.
-	if CCKey == "" {
-		return "", errors.New("CCKey cannot be an empty string.")
+	config.CCKey = os.Getenv("CCKEY")
+	config.URL = os.Getenv("APIURL")
+	if config.CCKey == "" {
```

Comment:
This is incorrect, I think. Looks like you are only checking if CCKey is empty and not if URL is empty.